tensorflow/python/ops/image_ops_test.py

```diff
@@ -768,6 +789,33 @@ def testRandomFlipLeftRight(self):
       self.assertGreaterEqual(count_flipped, 1)
       self.assertGreaterEqual(count_unflipped, 1)
 
+  def testRandomFlipLeftRightWithBatch(self):
+    x_np = np.array([[[1, 2, 3], [1, 2, 3]], [[1, 2, 3], [1, 2, 3]]],
+                    dtype=np.uint8).reshape([2, 2, 3, 1])
+    y_np = np.array([[[3, 2, 1], [3, 2, 1]], [[3, 2, 1], [3, 2, 1]]],
+                    dtype=np.uint8).reshape([2, 2, 3, 1])
+
+    with self.test_session(use_gpu=True):
+      x_tf = constant_op.constant(x_np, shape=x_np.shape).eval()
+      y = image_ops.random_flip_left_right(x_tf)
+      count_flipped = 0
+      count_unflipped = 0
+      for _ in range(50):
+        y_tf = y.eval()
+        for index in range(0, x_tf.shape[0]):
+          current_x_tf = x_tf[index]
+          current_y_tf = y_tf[index]
+          current_y_np = y_np[index]
+
+          if current_y_tf[0][0] == 1:
+            self.assertAllEqual(current_y_tf, current_x_tf)
+            count_unflipped += 1
+          else:
+            self.assertAllEqual(current_y_tf, current_y_np)
+            count_flipped += 1
+      self.assertGreaterEqual(count_flipped, 1)
+      self.assertGreaterEqual(count_unflipped, 1)
```

Comment:
The number should be binomially distributed around 50, so the stddev of the normal approximation is 5. We should be pretty safe to assert that count_flipped > 20 and count_unflipped > 20. That's still 6 sigma.